cal/cal/models.py

```diff
@@ -584,6 +584,21 @@ def query(self, calendar_ids=None, start=None, end=None):
         return events_qs
 
 
+    def queryWeek(self, calendar_ids=None, start=None, end=None):
+        queryset = []
+        events = self.query().order_by('start')
+        i = 0
+        start = events[0].start # We have to start with this outside for loop, seems sketch
+        while i < len(events):
+            end = start + timedelta(days=7)
+            total = 0
+            while i < len(events) and (end - events[i].start).total_seconds() >= 0:
+                total += (events[i].end - events[i].start).total_seconds() / 3600
```

Comment:
No can do, it has to be in the while condition as the difference is changing each iteration.